calico_kubernetes/calico_kubernetes.py

```diff
@@ -725,22 +755,26 @@ def main():
         pod_name = sys.argv[3].replace('/', '_')
         docker_id = sys.argv[4]
 
-        logger.info("Using LOG_LEVEL=%s", LOG_LEVEL)
-        logger.info("Using ETCD_AUTHORITY=%s", os.environ[ETCD_AUTHORITY_ENV])
-        logger.info("Using CALICOCTL_PATH=%s", CALICOCTL_PATH)
-        logger.info("Using KUBE_API_ROOT=%s", KUBE_API_ROOT)
-        logger.info("Using CALICO_IPAM=%s", CALICO_IPAM)
-        logger.info("Using DEFAULT_POLICY=%s", DEFAULT_POLICY)
-
-        if mode == 'setup':
-            logger.info('Executing Calico pod-creation hook')
-            NetworkPlugin().create(namespace, pod_name, docker_id)
-        elif mode == 'teardown':
-            logger.info('Executing Calico pod-deletion hook')
-            NetworkPlugin().delete(namespace, pod_name, docker_id)
-        elif mode == 'status':
+        if mode == "status":
+            # Status is called on a regular basis - handle separately
+            # to avoid flooding the logs.
             logger.info('Executing Calico pod-status hook')
             NetworkPlugin().status(namespace, pod_name, docker_id)
+        else:
+            logger.info("Using LOG_LEVEL=%s", LOG_LEVEL)
```

Comment:
Worth mentioning that INIT is going away in CNI, so moving stuff over there would be temporary.

From: Casey Davenport [mailto:notifications@github.com]
Sent: Monday, October 05, 2015 3:53 PM
To: projectcalico/calico-kubernetes
Cc: Paul Tiplady
Subject: Re: [calico-kubernetes] Log INFO and above to stdout (#60)


In calico_kubernetes/calico_kubernetes.py<https://github.com/projectcalico/calico-kubernetes/pull/60#discussion_r41208190>:

>              logger.info('Executing Calico pod-status hook')

>              NetworkPlugin().status(namespace, pod_name, docker_id)

> +        else:

> +            logger.info("Using LOG_LEVEL=%s", LOG_LEVEL)

I'd kind of like to print them as info logs during init and then as debug logs other time.

On that note, it might be worthwhile to do some validation in init - e.g check we can reach the KUBE_API_ROOT and ETCD_AUTHORITY. Might make debugging easier, since the kubelet will fail to start if the plugin is misconfigured (rather than starting but failing the create pods). Just a thought.

—
Reply to this email directly or view it on GitHub<https://github.com/projectcalico/calico-kubernetes/pull/60/files#r41208190>.